code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -289,66 +290,46 @@
 	attack_verb_simple = "kick"
 	health = 15
 	maxHealth = 15
-	var/eggsleft = 0
-	var/eggsFertile = TRUE
-	var/body_color
-	var/icon_prefix = "chicken"
 	pass_flags = PASSTABLE | PASSMOB
 	mob_size = MOB_SIZE_SMALL
-	var/list/feedMessages = list("It clucks happily.","It clucks happily.")
-	var/list/layMessage = EGG_LAYING_MESSAGES
-	var/list/validColors = list("brown","black","white")
 	gold_core_spawnable = FRIENDLY_SPAWN
-	var/static/chicken_count = 0
-
 	footstep_type = FOOTSTEP_MOB_CLAW
+	///counter for how many chickens are in existence to stop too many chickens from lagging shit up
+	var/static/chicken_count = 0
+	///boolean deciding whether eggs laid by this chicken can hatch into chicks
+	var/process_eggs = TRUE
 
 /mob/living/simple_animal/chicken/Initialize()
 	. = ..()
-	if(!body_color)
-		body_color = pick(validColors)
-	icon_state = "[icon_prefix]_[body_color]"
-	icon_living = "[icon_prefix]_[body_color]"
-	icon_dead = "[icon_prefix]_[body_color]_dead"
-	pixel_x = base_pixel_x + rand(-6, 6)
-	pixel_y = base_pixel_y + rand(0, 10)
-	++chicken_count
+	chicken_count++
 	add_cell_sample()
-
+	AddElement(/datum/element/animal_variety, "chicken", pick("brown","black","white"), TRUE)
+	var/list/feed_messages = list("[p_they()] clucks happily.")
+	var/eggs_left = 0
+	var/eggs_added_from_eating = rand(1, 4)
+	var/max_eggs_held = 8
+	AddComponent(/datum/component/egg_layer,\
+		/obj/item/food/egg,\
+		list(/obj/item/food/grown/wheat),\
+		feed_messages,\
+		EGG_LAYING_MESSAGES,\
+		eggs_left,\
+		eggs_added_from_eating,\
+		max_eggs_held,\
+		.proc/egg_laid\
```

Comment:
You're expecting a callback, not a proc path. This is going to runtime.

Reply:
good catch